File: challenges/elixir/lib/library_data.ex
defmodule LibraryData do
  def library_data do
    %{
      "address" => "Here and now",
      "catalog" => %{
        "authorsById" => %{
          "alan-moore" => %{
            "bookIsbns" => ["978-1779501127"],
            "name" => "Alan Moore"
          },
          "dave-gibbons" => %{
            "bookIsbns" => ["978-1779501127"],
            "name" => "Dave Gibbons"
          }
        },
        "booksByIsbn" => %{
          "978-1779501127" => %{
            "authorIds" => ["alan-moore", "dave-gibbons"],
            "bookItems" => [
              %{"id" => "book-item-1", "rackId" => "rack-17"},
              %{"id" => "book-item-2", "rackId" => "rack-17"}
            ],
            "isbn" => "978-1779501127",
            "publicationYear" => 1987,
            "title" => "Watchmen"
          }
        }
      },
      "name" => "The smallest library on earth",
      "userManagement" => %{
        "librarians" => %{
          "franck@gmail.com" => %{
            "email" => "franck@gmail.com",
            "encryptedPassword" => "bXlwYXNzd29yZA=="
          }
        },
        "members" => %{
          "samantha@gmail.com" => %{
            "email" => "samantha@gmail.com",
            "encryptedPassword" => "c2VjcmV0",
            "isBlocked" => false
          }
        }
      }
    }
  end
end

File: challenges/c#/Models/LibraryData.cs
using System.Collections.Immutable;

namespace Models;
public static class Model
{
    public static readonly ImmutableDictionary<string, object> LibraryData = new Dictionary<string, object>()
    {
        { "name", "The smallest library on earth" },
        { "address", "Here and now" },
        { "catalog", new Dictionary<string, object>()
            {
                {
                    "booksByIsbn", new Dictionary<string, object>()
                    {
                        {
                            "978-1779501127", new Dictionary<string, object>()
                            {
                                { "isbn", "978-1779501127" },
                                { "title", "Watchmen" },
                                { "publicationYear", 1987 },
                                { "authorIds", new string[] { "alan-moore", "dave-gibbons" } },
                                { "bookItems", new (string id, string rackID)[] { ("book-item-1", "rack-17"), ("book-item-2", "rack-17") } }
                            }
                        }
                    }
                }
            }
        },
        { "authorsById", new Dictionary<string, object>()
            {
                {
                    "alan-moore", new Dictionary<string, object>()
                    {
                        { "name", "Alan Moore" },
                        { "booksIsbns", new string[] {"978-1779501127" } }
                    }
                },
                {
                    "dave-gibbons", new Dictionary<string, object>()
                    {
                        { "name", "Dave Gibbons" },
                        { "booksIsbns", new string[] {"978-1779501127" } }
                    }
                }
            }
        },
        { "userManagement", new Dictionary<string, object>()
            {
                { 
                    "librarians", new Dictionary<string, object>()
                    {
                        {
                            "franck@gmail.com", new Dictionary<string, object>()
                            {
                                { "email", "franck@gmail.com" },
                                { "encryptedPassword", "bXlwYXNzd29yZA==" }
                            }
                        },

                    }
                },
                {
                    "members", new Dictionary<string, object>()
                    {
                        {
                            "samantha@gmail.com", new Dictionary<string, object>()
                            {
                                { "email", "samantha@gmail.com" },
                                { "encryptedPassword", "c2VjcmV0" },
                                { "isBlocked", false }
                            }
                        },
                    }
                }
            }
        }
    }.ToImmutableDictionary();
}
